stop_controlled_intersection_tactical_plugin/include/stop_controlled_intersection_plugin.hpp

```diff
@@ -157,30 +138,38 @@ class StopControlledIntersectionTacticalPlugin
    * 
    * \return A list of trajectory points to send to the carma planning stack
    */ 
-  std::vector<cav_msgs::TrajectoryPlanPoint> compose_trajectory_from_centerline(
-    const std::vector<PointSpeedPair>& points, const cav_msgs::VehicleState& state, const ros::Time& state_time); 
-  
-    
-    /**
-   * \brief Method to call at fixed rate in execution loop. Will publish plugin discovery updates
-   * 
-   * \return True if the node should continue running. False otherwise
-   */ 
-  bool onSpin();
+  std::vector<carma_planning_msgs::msg::TrajectoryPlanPoint> compose_trajectory_from_centerline(
+    const std::vector<PointSpeedPair>& points, const carma_planning_msgs::msg::VehicleState& state, const rclcpp::Time& state_time); 
+
+  // overrides
+  carma_ros2_utils::CallbackReturn on_configure_plugin() override;
+  bool get_availability();
+  std::string get_version_id();
+  rcl_interfaces::msg::SetParametersResult 
+  parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters);
+  void plan_trajectory_callback(
+    std::shared_ptr<rmw_request_id_t>, 
+    carma_planning_msgs::srv::PlanTrajectory::Request::SharedPtr req, 
+    carma_planning_msgs::srv::PlanTrajectory::Response::SharedPtr resp) override;
 
   private:
 
   carma_wm::WorldModelConstPtr wm_;
   StopControlledIntersectionTacticalPluginConfig config_;
-  PublishPluginDiscoveryCB plugin_discovery_publisher_;
- 
-  cav_msgs::Plugin plugin_discovery_msg_;
-  carma_debug_msgs::TrajectoryCurvatureSpeeds debug_msg_;
+
+  // Unit test helper functions
+  carma_wm::WorldModelConstPtr get_wm() { return wm_; }
```

Comment:
Why do we need these? Can't we just access the wm_ pointer?

Reply:
somehow I wasn't able to access wm_ before even if I added FRIEND_TEST, but it does work now. Maybe I had a typo. I removed these helper functions now.